local/item.go

```diff
@@ -27,19 +27,24 @@ const (
 )
 
 type item struct {
-	path     string
-	infoOnce sync.Once // protects info
-	info     os.FileInfo
-	infoErr  error
-	metadata map[string]interface{}
+	relPath   string // OS-native relative path
+	container *container
```

Comment:
why put the entire container on the item instead of just its path? A lot of coupling here that seems unwarranted